### ISSUE TITLE
❗Different Results Between Python and iOS (CoreML) — Ultralytics YOLO → CoreML → yolo-ios-app

### ISSUE THREAD

**Issue by HansolWorld** [ultralytics/yolo-ios-app#208]
Hi, I trained a model using Ultralytics YOLO, exported it to CoreML (.mlpackage), and confirmed that inference results from Python (.pt and .mlpackage) are identical.
However, when I use the same .mlpackage model on iOS (using the official yolo-ios-app example), the results are completely different.

🧠 Summary
	•	✅ Python: best.pt and best.mlpackage produce identical predictions
	•	❌ iOS (yolo-ios-app): using the same .mlpackage gives very different results
	•	🖼️ Tested with the same image from the photo library

🧩 Steps to Reproduce

1️⃣ Train
```python
from ultralytics import YOLO

model = YOLO("yolo11n.yaml")  # or yolo11n.pt
results = model.train(
    data="datasets/version12/data.yaml",
    epochs=50,
    imgsz=640,
    batch=-1,
    workers=1,
    perspective=0.01,
    translate=0,
    scale=0,
    mosaic=0,
    erasing=0,
    pretrained=True,
    rect=True,
)
```

2️⃣ Export to CoreML
```python
from ultralytics import YOLO

def export_and_zip_yolo_models():
    imgsz = 640
    model_name = "detect/train8/weights/best"
    model = YOLO(f"{model_name}.pt")
    model.export(format="coreml", imgsz=imgsz, conf=0.4, iou=0.3, nms=False)
```

3️⃣ Compare .pt vs .mlpackage in Python (✅ identical results)
```python
from ultralytics import YOLO

device = 'cuda' if torch.cuda.is_available() else \
         'mps' if torch.backends.mps.is_available() else 'cpu'

model_pt = YOLO("detect/train8/weights/best.pt")
model_ml = YOLO("detect/train8/weights/best.mlpackage")

results_ml = model_ml.predict(
    source,
    imgsz=640,
    conf=0.4,
    device=device,
    agnostic_nms=True,
    iou=0.3,
)

results_pt = model_pt.predict(
    source,
    imgsz=640,
    conf=0.4,
    device=device,
    agnostic_nms=True,
    iou=0.3,
)

# Both results are identical (same boxes/classes/scores)
```
4️⃣ Run the same .mlpackage on iOS (❌ different results)

Using the yolo-ios-app SwiftUI example:
```swift
struct ContentView: View {
    @State private var selectedItem: PhotosPickerItem?
    @State private var inputImage: UIImage?
    @State private var yoloResult: YOLOResult?

    let yolo = YOLO("best", task: .detect) // best.mlpackage

    var body: some View {
        VStack {
            if let annotated = yoloResult?.annotatedImage {
                Image(uiImage: annotated).resizable().scaledToFit()
            } else if let input = inputImage {
                Image(uiImage: input).resizable().scaledToFit()
            } else {
                Text("No image selected")
            }

            PhotosPicker(selection: $selectedItem, matching: .images) {
                Text("Pick Photo")
                    .padding().foregroundColor(.white)
                    .background(Color.blue).cornerRadius(8)
            }
            .onChange(of: selectedItem) { newItem in
                Task {
                    guard let newItem = newItem,
                          let data = try? await newItem.loadTransferable(type: Data.self),
                          let uiImage = UIImage(data: data) else { return }

                    let correct = getCorrectOrientationUIImage(uiImage: uiImage)
                    inputImage = correct
                    yoloResult = yolo(correct) // Results differ from Python
                    print(yoloResult as Any)
                }
            }
        }
        .padding()
    }
}

func getCorrectOrientationUIImage(uiImage: UIImage) -> UIImage {
    var newImage = UIImage()
    let ciContext = CIContext()
    switch uiImage.imageOrientation.rawValue {
    case 1:
        guard let oriented = CIImage(image: uiImage)?.oriented(.down),
              let cg = ciContext.createCGImage(oriented, from: oriented.extent) else { return uiImage }
        newImage = UIImage(cgImage: cg)
    case 3:
        guard let oriented = CIImage(image: uiImage)?.oriented(.right),
              let cg = ciContext.createCGImage(oriented, from: oriented.extent) else { return uiImage }
        newImage = UIImage(cgImage: cg)
    default:
        newImage = uiImage
    }
    return newImage
}
```

✅ Expected Behavior

The predictions from iOS (yolo-ios-app) should closely match those from Python (both .pt and .mlpackage).

❌ Actual Behavior
	•	Bounding boxes, class labels, and confidence scores are significantly different on iOS.
	•	The same image, imgsz=640, conf=0.4, and iou=0.3 were used.

❓What could be the reason for this difference?
